core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1759,61 +1765,72 @@ public void dropRole(Session session, String role, Optional<String> catalog)
         CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalog.get());
         CatalogName catalogName = catalogMetadata.getCatalogName();
         ConnectorMetadata metadata = catalogMetadata.getMetadata();
-
         metadata.dropRole(session.toConnectorSession(catalogName), role);
     }
 
     @Override
     public Set<String> listRoles(Session session, Optional<String> catalog)
     {
-        if (catalog.isEmpty()) {
-            return ImmutableSet.of();
+        if (catalog.isPresent()) {
```

Comment:
inverse the condition, here all below

Reply:
See the comment, this code falls through to the system metadata.  These listing commands are used by information schema, and we want information schema still show roles when the catalog is using system security (it will show system roles instead of nothing).